src/uwtools/tests/test_rocoto.py

```diff
@@ -1,117 +1,220 @@
-    assert rocoto._rocoto_schema_xml() == expected
 
+# Tests
 
-@pytest.mark.parametrize("vals", [("hello_workflow.yaml", True), ("fruit_config.yaml", False)])
-def test_realize_rocoto_xml(vals, tmp_path):
-    fn, validity = vals
-    output = tmp_path / "rendered.xml"
 
-    with patch.object(rocoto, "validate_rocoto_xml", value=True):
-        with resources.as_file(resources.files("uwtools.tests.fixtures")) as path:
-            config_file = path / fn
-            result = rocoto.realize_rocoto_xml(config_file=config_file, output_file=output)
-    assert result is validity
+def test_realize_rocoto_xml_to_file(assets):
+    cfgfile, outfile = assets
+    assert rocoto.realize_rocoto_xml(config_file=cfgfile, output_file=outfile) is True
 
 
-def test_realize_rocoto_default_output():
-    with patch.object(rocoto, "validate_rocoto_xml", value=True):
-        with resources.as_file(resources.files("uwtools.tests.fixtures")) as path:
-            config_file = path / "hello_workflow.yaml"
-            result = rocoto.realize_rocoto_xml(config_file=config_file)
-    assert result is True
+def test_realize_rocoto_xml_to_stdout(capsys, assets):
+    cfgfile, outfile = assets
+    assert rocoto.realize_rocoto_xml(config_file=cfgfile) is True
+    with open(outfile, "w", encoding="utf-8") as f:
+        f.write(capsys.readouterr().out)
+    assert rocoto.validate_rocoto_xml(outfile)
 
 
-def test_realize_rocoto_invalid_xml():
-    config_file = support.fixture_path("hello_workflow.yaml")
-    xml = support.fixture_path("rocoto_invalid.xml")
-    with patch.object(rocoto, "_write_rocoto_xml", return_value=None):
-        with patch.object(tempfile, "mkstemp", return_value=(None, xml)):
-            result = rocoto.realize_rocoto_xml(config_file=config_file, output_file=xml)
-    assert result is False
+def test_realize_rocoto_invalid_xml(assets):
+    cfgfile, outfile = assets
+    dump = lambda _, dst: shutil.copyfile(fixture_path("rocoto_invalid.xml"), dst)
+    with patch.object(rocoto._RocotoXML, "dump", dump):
+        assert rocoto.realize_rocoto_xml(config_file=cfgfile, output_file=outfile) is False
 
 
 @pytest.mark.parametrize("vals", [("hello_workflow.xml", True), ("rocoto_invalid.xml", False)])
-def test_rocoto_xml_is_valid(vals):
+def test_validate_rocoto_xml(vals):
     fn, validity = vals
-    xml = support.fixture_path(fn)
-    result = rocoto.validate_rocoto_xml(input_xml=xml)
-
-    assert result is validity
-
-
-def test__write_rocoto_xml(tmp_path):
-    config_file = support.fixture_path("hello_workflow.yaml")
-    output = tmp_path / "rendered.xml"
-
-    rocoto._write_rocoto_xml(config_file=config_file, output_file=output)
-
-    expected = support.fixture_path("hello_workflow.xml")
-    assert support.compare_files(expected, output) is True
+    xml = fixture_path(fn)
+    assert rocoto.validate_rocoto_xml(input_xml=xml) is validity
+
+
+def test__RocotoXML__doctype_entities(instance):
```

Comment:
Good question. I don't know. Maybe, since the tests themselves map onto a class, and because they have a specific fixture they all share that has nothing to do with the tests for the standalone module functions. I'm going to google for "when to use classes in pytest" and see what I learn. I may circle back and refactor this in a later PR if it seems like that would be a good idea.

Well, as I was typing that, one thing came to mind: The presence of `_RocotoXML` in the names of each of these tests would probably go away if they were in a class of their own. That alone seems to me to argue for that approach.